### PR TITLE
Add the output fields of wallet list subcommand

### LINKED ISSUE #1778: Add the output fields of wallet subcommand list
**Issue summary**

Add output fields "Default" and "Balance" to cli wallet list
```sh
❯ ./forest wallet list
Address                                    Default  Balance
t1ss6z757xy46h2qhxgnk4rummmjwmpbtwcnkdcki  X        0
```

**Other information and links**

### PR DESCRIPTION
**Pull request by jimluo**
**Summary of changes**
Changes introduced in this pull request:
- 

Added  new fields "Default"  and "Balance" to wallet list subcommand

**Reference issue to close (if applicable)**

Closes 
#1778

**Other information and links**

### CODE REVIEW

File: forest/src/cli/wallet_cmd.rs [hunk 1]
```diff
@@ -165,10 +165,23 @@ impl WalletCommands {
...
                 let response = wallet_list().await.map_err(handle_rpc_err).unwrap();
+                let default = wallet_default_address()
+                    .await
+                    .map_err(handle_rpc_err)
+                    .unwrap();
 
-                response.iter().for_each(|address| {
-                    println!("{}", address.0);
-                });
+                println!("Address                                    Default  Balance");
```
Comment (lemmih) [inline forest/src/cli/wallet_cmd.rs:173]: The alignment could be done through `println`'s padding. Not a big deal, though.

**Review by lemmih — APPROVED** [review]
Looks okay to me.

File: forest/src/cli/wallet_cmd.rs [hunk 2]
```diff
@@ -165,10 +165,23 @@ impl WalletCommands {
...
+
+                for address in response {
+                    let addr = address.0.to_string();
+                    let def = if addr == default { "X" } else { "" };
+                    let bal = wallet_balance((addr.clone(),))
+                        .await
+                        .map_err(handle_rpc_err)
+                        .unwrap();
```
Comment (LesnyRumcajs) [inline forest/src/cli/wallet_cmd.rs:181]: `unwrap` in general is not a nice way to handle errors, would be great if you could change it to e.g. `expect("failed loading the wallet balance")` or something along those lines. At the same time, I can see that there are a bunch of unwraps across the file so it's just a suggestion to make it a bit better.

File: forest/src/cli/wallet_cmd.rs [hunk 3]
```diff
@@ -165,10 +165,23 @@ impl WalletCommands {
...
+                    .unwrap();
 
-                response.iter().for_each(|address| {
-                    println!("{}", address.0);
-                });
+                println!("Address                                    Default  Balance");
+
+                for address in response {
+                    let addr = address.0.to_string();
+                    let def = if addr == default { "X" } else { "" };
```
Comment (LesnyRumcajs) [inline forest/src/cli/wallet_cmd.rs:177]: `def` is kind of broad, how about `default_address_mark`?

File: forest/src/cli/wallet_cmd.rs [hunk 4]
```diff
@@ -165,10 +165,23 @@ impl WalletCommands {
...
+                    let bal = wallet_balance((addr.clone(),))
+                        .await
+                        .map_err(handle_rpc_err)
+                        .unwrap();
+
+                    println!("{}  {:7}  {}", addr, def, bal);
```
Comment (LesnyRumcajs) [inline forest/src/cli/wallet_cmd.rs:183]:
```suggestion
                    println!("{addr}  {def:7}  {bal}");
```

Comment (LesnyRumcajs) [inline forest/src/cli/wallet_cmd.rs:183]: I would like to see this to be aligned for sure with the header. E.g. `{addr:<the entire length of address column as in header>}{def:<the same>}{bal:<the same>}`. Right now, looking at the code it's hard to say if all will fit.

**Review by LesnyRumcajs — CHANGES_REQUESTED** [review]
Thanks for the contribution! Most things are details, but I would really like to see programmatically aligning the table.

File: forest/src/cli/wallet_cmd.rs (hunk 2 above)
Comment (lemmih) [reply forest/src/cli/wallet_cmd.rs:181]: The `unwrap` will never fail (`handle_rpc_err` returns `!`). We really want something like `into_ok` here if it was stable. Or we could refactor the code.

Comment (LesnyRumcajs) [reply forest/src/cli/wallet_cmd.rs:181]: I see, it makes sense then. We should probably consider refactoring this in the future, I wouldn't expect `map_` to have such side effects.

**Comment by jimluo** [comment]
Thank you for your advice, I will follow the advice soon to improve.

**Review by tyshko5 — APPROVED** [review]

File: forest/src/cli/wallet_cmd.rs [hunk 5]
```diff
@@ -165,10 +166,28 @@ impl WalletCommands {
...
+                    let default_address_mark = if addr == default { "X" } else { "" };
+                    let balance_string = wallet_balance((addr.clone(),))
+                        .await
+                        .map_err(handle_rpc_err)
+                        .expect("Failed loading the wallet balance");
+                    let balance = balance_string
+                        .parse::<BigInt>()
+                        .expect("Couldn't convert balance string to bigint");
+                    let balance_fil =
+                        balance_to_fil(balance).expect("Couldn't convert balance to fil");
```
Comment (LesnyRumcajs) [inline forest/src/cli/wallet_cmd.rs:187]:
```suggestion
                        balance_to_fil(balance).expect("Couldn't convert balance to FIL");
```

File: forest/src/cli/wallet_cmd.rs [hunk 6]
```diff
@@ -165,10 +166,28 @@ impl WalletCommands {
...
+                for address in response {
+                    let addr = address.0.to_string();
+                    let default_address_mark = if addr == default { "X" } else { "" };
+                    let balance_string = wallet_balance((addr.clone(),))
+                        .await
+                        .map_err(handle_rpc_err)
+                        .expect("Failed loading the wallet balance");
+                    let balance = balance_string
+                        .parse::<BigInt>()
+                        .expect("Couldn't convert balance string to bigint");
```
Comment (LesnyRumcajs) [inline forest/src/cli/wallet_cmd.rs:185]:
```suggestion
                        .expect("Couldn't convert balance string to BigInt");
```

Comment (LesnyRumcajs) [inline forest/src/cli/wallet_cmd.rs:185]: let's print the error reason and the value provided that caused the failure

File: forest/src/cli/wallet_cmd.rs (hunk 5 above)
Comment (LesnyRumcajs) [inline forest/src/cli/wallet_cmd.rs:187]: let's print the error reason and the value provided that caused the failure

File: forest/src/cli/wallet_cmd.rs [hunk 7]
```diff
@@ -165,10 +166,28 @@ impl WalletCommands {
...
                 let response = wallet_list().await.map_err(handle_rpc_err).unwrap();
+                let default = wallet_default_address()
+                    .await
+                    .map_err(handle_rpc_err)
+                    .unwrap();
 
-                response.iter().for_each(|address| {
-                    println!("{}", address.0);
-                });
+                println!("Address                                    Default  Balance");
```
Comment (LesnyRumcajs) [inline forest/src/cli/wallet_cmd.rs:174]: @jimluo let's add the same explicit padding like for the data row

**Review by LesnyRumcajs — CHANGES_REQUESTED** [review]

**Review by LesnyRumcajs — APPROVED** [review]

File: forest/src/cli/wallet_cmd.rs (hunk 7 above)
Comment (jimluo) [reply forest/src/cli/wallet_cmd.rs:174]: Thanks, I will fix it over the weekend

**Pull request merged**